### ISSUE TITLE
DataStax Enterprise configuration file (dse.yaml)

### ISSUE THREAD

**Issue by dallinb** [locp/cassandra#366]
https://docs.datastax.com/en/datastax_enterprise/4.6/datastax_enterprise/reference/refDseYaml.html

**Comment by samyray** [comment]
Please, please, please, add option for this file to be edited:

* /etc/dse/dse-env.sh

```
#!/bin/sh

# add any environment overrides you need here. This is where users
# may set thirdparty variables like HADOOP_LOG_DIR etc.
# Variables must be exported in order to be picked up, like:
# export HADOOP_LOG_DIR=<log_dir>

# This is for Spark scripts - this setting will modify their original behaviour
export DSE_MODE="1"

# This is here so the installer can force set DSE_HOME
#DSE_HOME="/usr/share/dse"
#export DSE_HOME

# ==================================
# don't change after this.
if [ "$DSE_HOME" = "" ]; then
  if [ -x "/usr/share/dse" ]; then
    DSE_HOME="/usr/share/dse"
    export DSE_HOME
  fi
fi

if [ -r "`dirname "$0"`/dse.in.sh" ]; then
    # File is right where the executable is
    . "`dirname "$0"`/dse.in.sh"
elif [ -r "/usr/share/dse/dse.in.sh" ]; then
    # Package install location
    . "/usr/share/dse/dse.in.sh"
elif [ -r "$DSE_HOME/bin/dse.in.sh" ]; then
    # Package install location
    . "$DSE_HOME/bin/dse.in.sh"
else
    # Go up directory tree from where we are and see if we find it
    DIR="`dirname $0`"
    for i in 1 2 3 4 5 6; do
        if [ -r "$DIR/bin/dse.in.sh" ]; then
            DSE_IN="$DIR/bin/dse.in.sh"
            break
        fi
        DIR="$DIR/.."
    done
    if [ -r "$DSE_IN" ]; then
        . "$DSE_IN"
    else
        echo "Cannot determine location of dse.in.sh"
        exit 1
    fi
fi
```


* /etc/dse/dserc-env.sh

```
#!/bin/sh

############################################
# Pull in ~/.dserc if it exists
# and extract the credentials
############################################
HADOOP_CREDENTIALS=""
HIVE_CREDENTIALS=""
SQOOP_CREDENTIALS=""
SPARK_CREDENTIALS=""
SHARK_CREDENTIALS=""
DSE_CREDENTIALS=""

set_credentials() {
    if [ -z $DSE_CREDENTIALS_SUPPLIED ]; then

        if [ ! -z $dse_username ]; then
            DSE_USERNAME="$dse_username"
            DSE_PASSWORD="$dse_password"
        elif [ -f ~/.dserc ]; then
            DSE_USERNAME=$(echo `cat ~/.dserc | grep username` | awk  '{ string=substr($0, index($0, "=") + 1); print string; }' )
            DSE_PASSWORD=$(echo `cat ~/.dserc | grep password` | awk  '{ string=substr($0, index($0, "=") + 1); print string; }' )
            DSE_SASL_PROTOCOL=$(echo `cat ~/.dserc | grep sasl_protocol` | awk  '{ string=substr($0, index($0, "=") + 1); print string; }' )
            DSE_LOGIN_CONFIG=$(echo `cat ~/.dserc | grep login_config` | awk  '{ string=substr($0, index($0, "=") + 1); print string; }' )
        fi

        if [ ! -z $DSE_USERNAME ]; then
            HADOOP_CREDENTIALS="-Dcassandra.username=$DSE_USERNAME -Dcassandra.password=$DSE_PASSWORD"
            HIVE_CREDENTIALS="--hiveconf cassandra.username=$DSE_USERNAME --hiveconf cassandra.password=$DSE_PASSWORD"
            SQOOP_CREDENTIALS="--cassandra-username=$DSE_USERNAME --cassandra-password=$DSE_PASSWORD"
            export SPARK_CREDENTIALS="$HADOOP_CREDENTIALS"
            export SHARK_CREDENTIALS="$HIVE_CREDENTIALS"
            export SPARK_SQL_SERVER_CREDENTIALS="$HIVE_CREDENTIALS"
            DSE_CREDENTIALS="$HADOOP_CREDENTIALS"
        fi
    fi
    if [ ! -z $dse_jmx_username ]; then
        DSE_JMX_USERNAME="$dse_jmx_username"
        DSE_JMX_PASSWORD="$dse_jmx_password"
    elif [ -f ~/.dserc ]; then
        DSE_JMX_USERNAME=$(echo `cat ~/.dserc | grep jmx_username` | awk  '{ string=substr($0, index($0, "=") + 1); print string; }' )
        DSE_JMX_PASSWORD=$(echo `cat ~/.dserc | grep jmx_password` | awk  '{ string=substr($0, index($0, "=") + 1); print string; }' )
    fi
    if [ ! -z $DSE_JMX_USERNAME ]; then
        DSE_JMX_CREDENTIALS="-a $DSE_JMX_USERNAME -b $DSE_JMX_PASSWORD"
        CASSANDRA_JMX_CREDENTIALS="-u $DSE_JMX_USERNAME -pw $DSE_JMX_PASSWORD"
    fi
}

set_credentials
```

**Comment by dallinb** [comment]
@samyray Please take a look at #367.  If you like what you see I'll back port that into 1.X of the module.

**Comment by dallinb** [comment]
At the request of @samyray this has been released in the legacy branch as 1.27.0.  It will be released in 2.4.0.
